packages/core/src/Agentica.ts

```diff
@@ -140,13 +158,16 @@ export class Agentica {
       text: content,
     };
     await this.dispatch(prompt);
+    const context = this.getContext({ prompt, usage: this.token_usage_ });
 
-    const newbie: IAgenticaPrompt[] = await this.executor_(
-      this.getContext({
-        prompt,
-        usage: this.token_usage_,
-      }),
-    );
+    const middlewares = this.middlewares_.get("conversate" as const);
+    if (middlewares) {
+      await Array.from(middlewares).reduce((acc, cur) => {
+        return this.middlewareCompose(acc, cur);
+      })(context, async () => {});
+    }
+
+    const newbie: IAgenticaPrompt[] = await this.executor_(context);
```

Comment:
middleware should cover reqeust and response
but, current architecture can't cover response  

nextjs:
https://nextjs.org/docs/app/building-your-application/routing/middleware

if user can use `next()`, it means cover request and response

Reply:
I just added a comment about this.

Reply:
the reason why we chose `next()` is because of to manage before process and after process.
however, the current middleware runs before the process.

For the previous process, we should not use the `next()` keyword as it confuses clients.